### DISCUSSION TITLE
Is it possible for a running task to get its own ID?

### DISCUSSION THREAD

**Discussion by zmarffy** [taskiq-python/taskiq#213] [Q&A] [answered]
I have a Postgres database as my backend. I would like to write the progress of the current task to the database (just a string that periodically updates "where the task is" in its function code). To do so, I would think I would need to write to the database result backend said string. But to do so I would need my task ID (and this goes without saying that the backend's `set_result` function would do an update, not an insert, and make a middleware to do the initial insert, which is fine). Is there a way to get the task's own ID so I can do this, or am I designing this awkwardly wrong?

**Comment by zmarffy** [comment, marked as answer]
Yes. I have used [`Context`](https://github.com/taskiq-python/taskiq/blob/34db231d49abf41d4cf1642eb04bfaef5d62dffe/taskiq/context.py#L11) to do so. Whether or not what I described is a good idea is still up for debate though.

**Reply by s3rius** [reply]
This is the exact solution to this problem. Thanks for posting it here.
